opendatahub-io/opendatahub-operator
# feat: Mapping component versions to RHOAI releases

`

## Description


The PR is a first step to the feature initiative tracked in [JIRA](https://issues.redhat.com/browse/RHOAISTRAT-327) and this PR adds feature to capture RHOAI release versions from all the components(that are supported by DSC) repositories and include it in the DSC status. 

JIRA: https://issues.redhat.com/browse/RHOAIENG-12619


## How Has This Been Tested?




To test with DevFlags:
- In a forked version of the components repository and create an .env file in config/manifest directory, have it in your branch.
- Point that forked repository as a DevFlag like [this](https://github.com/opendatahub-io/opendatahub-operator/wiki/4.-Using-Operator-in-Dev-Mode#devflags-example)
- Build and deploy the changes into the cluster.
- Create DSCI and DSC instances and you will be able see the status.components get updated with component upstream release details like below

`
apiVersion: [datasciencecluster.opendatahub.io/v1](http://datasciencecluster.opendatahub.io/v1)
kind: DataScienceCluster
metadata:

spec:
  components:
   
status:
 components:
  datasciencepipelines
   upstreamReleases:
    - name: datasciencepipelines
      displayName: Red Hat Build of Kubeflow Pipelines
      version: 2.2.0
      repoUrl: https://github.com/opendatahub-io/data-science-pipelines
 `

## Screenshot or short clip
<img width="985" alt="Screenshot 2024-10-26 at 12 12 59 PM" src="https://github.com/user-attachments/assets/81efaacd-8a36-41e5-85a2-d0e02dd6936c">




## Merge criteria
